database/linkeddb/linkeddb.go

```diff
@@ -281,7 +281,7 @@ func (ldb *linkedDB) getHeadKey() ([]byte, error) {
 func (ldb *linkedDB) putHeadKey(key []byte) error {
 	ldb.headKeyIsUpdated = true
 	ldb.updatedHeadKeyExists = true
-	ldb.updatedHeadKey = key
+	ldb.updatedHeadKey = slices.Clone(key)
```

Comment:
On mobile right now - but I think there are other instances in the linkeddb that we need to perform a copy

Reply:
In `Put` we hold a direct reference to `key` which can be problematic:
```golang
oldHead.Previous = key
```

Reply:
I think putting the copy in `putHeadKey` results in an unnecessary copy during `Delete` as well. Maybe we could copy the `key` in `Put` once we know we are going to hold a reference to it (aka after we check that we aren't overwriting an existing node)?

Reply:
updated